internal/pkg/aws/cloudwatch/cloudwatch.go

```diff
@@ -73,32 +73,53 @@ func (cw *CloudWatch) AlarmsWithTags(tags map[string]string) ([]AlarmStatus, err
 		}
 		alarmNames = append(alarmNames, name)
 	}
-	return cw.AlarmStatus(alarmNames)
+	return cw.AlarmStatuses(WithNames(alarmNames))
 }
 
-// AlarmStatus returns the status of each given alarm name.
-func (cw *CloudWatch) AlarmStatus(alarms []string) ([]AlarmStatus, error) {
-	if len(alarms) == 0 {
+// DescribeAlarmOpts sets the optional parameter for DescribeAlarms
+type DescribeAlarmOpts func(input *cloudwatch.DescribeAlarmsInput)
+
+// WithNames sets DescribeAlarms to filter on alarm names.
+func WithNames(names []string) DescribeAlarmOpts {
+	return func(in *cloudwatch.DescribeAlarmsInput) {
+		in.AlarmNames = aws.StringSlice(names)
+	}
+}
+
+// WithPrefix sets DescribeAlarms to filter on a name prefix.
+func WithPrefix(prefix string) DescribeAlarmOpts {
+	return func(in *cloudwatch.DescribeAlarmsInput) {
+		in.AlarmNamePrefix = aws.String(prefix)
+	}
+}
+
+// AlarmStatuses returns the status of each given alarm name.
+func (cw *CloudWatch) AlarmStatuses(opts ...DescribeAlarmOpts) ([]AlarmStatus, error) {
+	if len(opts) == 0 {
```

Comment:
If so why not make required ones in param? For example if name is required

```golang
AlarmStatuses(name string, opts ...DescribeAlarmOpts)
```

Reply:
Because we aren't searching for specific alarms by name; does that make sense?

Reply:
ah i see so this function takes at least one param. Should we return an error then?

Reply:
In theory there could be a future use case to return all statuses! 🤷🏼

Reply:
Yeah I think that'll make more sense to me too. Can we remove this condition for now?